devito/function.py

```diff
@@ -1203,8 +1242,13 @@ def _arg_values(self, **kwargs):
 
         return values
 
-    def _arg_apply(self, data):
-        self._dist_gather(data)
+    def _arg_apply(self, data, alias=None):
+        key = alias if alias is not None else self
+        if isinstance(key, AbstractSparseFunction):
+            key._dist_gather(data)
+        elif self.grid.distributor.nprocs > 1:
+            raise NotImplementedError("Don't know how to gather data from an "
```

Comment:
I wouldn't throw an error, only a waring, everything ran fine, just saying input was not changed and is to it's original value